lighthouse/helpers/plate_events.py

```diff
@@ -0,0 +1,343 @@
+import logging
+from typing import Optional, Dict, Tuple, List
+from uuid import uuid4
+from datetime import datetime
+from flask import current_app as app
+from lighthouse.messages.message import Message  # type: ignore
+from lighthouse.constants import (
+    PLATE_EVENT_SOURCE_COMPLETED,
+    PLATE_EVENT_SOURCE_NOT_RECOGNISED,
+    PLATE_EVENT_SOURCE_NO_MAP_DATA,
+    PLATE_EVENT_SOURCE_ALL_NEGATIVES,
+    FIELD_ROOT_SAMPLE_ID,
+    FIELD_RNA_ID,
+    FIELD_LAB_ID,
+    FIELD_RESULT,
+    FIELD_LH_SAMPLE_UUID,
+)
+from lighthouse.helpers.mongo_db import (
+    get_source_plate_uuid,
+    get_samples,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def construct_event_message(
+    event_type: str, params: Dict[str, str]
+) -> Tuple[List[str], Optional[Message]]:
+    """Delegates to the appropriate event construction method;
+    otherwise returns with errors for unknown event type.
+
+    Arguments:
+        event_type {str} -- The event type for which to construct a message.
+        params {Dict[str, str]} -- All parameters of the plate event message request.
+
+    Returns:
+        {[str]} -- Any errors attempting to construct the message, otherwise an empty array.
+        {Message} -- The constructed message; otherwise None if there are any errors.
+    """
+    if event_type == PLATE_EVENT_SOURCE_COMPLETED:
+        return construct_source_plate_completed_message(params)
+    elif event_type == PLATE_EVENT_SOURCE_NOT_RECOGNISED:
+        return construct_source_plate_not_recognised_message(params)
+    elif event_type == PLATE_EVENT_SOURCE_NO_MAP_DATA:
+        return construct_source_plate_no_map_data_message(params)
+    elif event_type == PLATE_EVENT_SOURCE_ALL_NEGATIVES:
+        return construct_source_plate_all_negatives_message(params)
+    else:
+        return [f"Unrecognised event type '{event_type}'"], None
+
+
+def get_routing_key(event_type: str) -> str:
+    """Determines the routing key for a plate event message.
+
+    Arguments:
+        event_type {str} -- The event type for which to determine a routing key
+
+    Returns:
+        {str} -- The message routing key.
+    """
+    return app.config["RMQ_ROUTING_KEY"].replace("#", event_type)
```

Comment:
I'm not sure we should have '#' in the variable if there is no use for us. Could we remove it and concat the event type instead?

Reply:
Sorry, I see is for the tests. Forget about the previous comment 👍